app/controllers/application_controller.rb

```diff
@@ -72,4 +72,11 @@ def encryptor
   def metrics_logger
     METRICS_LOGGER
   end
+
+  def ensure_visit_integrity
+    unless visit && visit.prisoner && visit.prisoner.prison_name.present?
+      redirect_to edit_prisoner_details_path,
+        notice: 'We’re missing some vital information'
```

Comment:
Ah Ok @threedaymonk Maybe: 

++
You need to complete missing information to start or continue your visit request
++ 

Hopefully that makes more sense?